packages/astro/src/core/create-vite.ts

```diff
@@ -91,8 +91,6 @@ export async function createVite(inlineConfig: ViteConfigWithSSR, { astroConfig,
 		}
 	}
 
-	// Add in user settings last, followed by any Vite configuration passed in from the parent function (overrides)
-	viteConfig = vite.mergeConfig(viteConfig, astroConfig.vite || {}); // merge in Vite config from astro.config.mjs
```

Comment:
Isn't this where we merge in any Vite configuration passed by users?

Reply:
In local testing, my user configuration still came thru, because it’s merged by [`createViteServer` for **dev**](https://github.com/withastro/astro/blob/814ffc5c6d678ad337ebde03c7f7b434fb03c88f/packages/astro/src/core/dev/index.ts#L236-L245) and [`build` for **preview**](https://github.com/withastro/astro/blob/814ffc5c6d678ad337ebde03c7f7b434fb03c88f/packages/astro/src/core/build/index.ts#L61-L70).

@natemoo-re, do you think we should flip this behavior, so that it’s only done in this file?

However, I think we may not have any tests for `vite` configurations passed in by users. I can add one.